doc/adr/0004-project-versioning.md

```diff
@@ -0,0 +1,29 @@
+# Project versioning with SNAPSHOT's
```

Comment:
name of ADR should correspond to the filename so either `0004-project-versioning-with-snapshots` or remove that part here